File: order/src/main/java/com/crm/order/service/OrderService.java
package com.crm.order.service;

import com.crm.order.entities.dto.request.OrderRequestDTO;
import com.crm.order.entities.dto.response.OrderResponseDTO;
import com.crm.order.entities.model.SearchCriteria;
import com.crm.order.enums.OrderStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface OrderService {
    OrderResponseDTO createOrder(OrderRequestDTO orderRequestDTO);
    OrderResponseDTO getOrderById(UUID id);
    Page<OrderResponseDTO> getAllOrdersWithCretiria(SearchCriteria SearchCretiria, Pageable pageable);

}

File: CRM-FRONT/crm/src/services/orderService.js
import axiosInstance from '../utils/axiosInstance';

const ORDER_API = '/orders';

const createOrder = async (orderData) => {
  try {
    const response = await axiosInstance.post(ORDER_API, orderData);
    return response.data;
  } catch (error) {
    throw error;
  }
};

const getOrderById = async (id) => {
  try {
    const response = await axiosInstance.get(`${ORDER_API}/${id}`);
    return response.data;
  } catch (error) {
    throw error;
  }
};

const searchOrders = async (searchCriteria, page = 0, size = 10) => {
  try {
    const params = {
      ...searchCriteria,
      page,
      size
    };
    const response = await axiosInstance.get(`${ORDER_API}/search`, { params });
    return response.data;
  } catch (error) {
    throw error;
  }
};

const orderService = {
  createOrder,
  getOrderById,
  searchOrders,
};

export default orderService;
